model/role.go

```diff
@@ -88,6 +95,36 @@ func (r *Role) Patch(patch *RolePatch) {
 	}
 }
 
+// MergeHigherScopedPermissions is meant to be invoked on a channel scheme's role and merges the higher-scoped
+// channel role's permissions.
+func (r *Role) MergeHigherScopedPermissions(higherScopedPermissions *RolePermissions) {
+	mergedPermissions := []string{}
+
+	for _, cp := range ALL_PERMISSIONS {
+		if cp.Scope != PERMISSION_SCOPE_CHANNEL {
+			continue
+		}
+
+		if higherScopedPermissions.RoleID == CHANNEL_ADMIN_ROLE_ID && slices.IncludesString(higherScopedPermissions.Permissions, cp.Id) {
```

Comment:
It may be worth converting `higherScopedPermissions.Permissions` into a map at the start of the method to avoid having to do so many loops since each call to `slices.IncludesString` is costing the length of `Permissions`

Reply:
minor optimization though since the length of Permissions is pretty small but I think it would be worth it

Reply:
Done.